proto/cosmos/distribution/query.proto

```diff
@@ -5,37 +5,56 @@ import "cosmos/query/pagination.proto";
 import "gogoproto/gogo.proto";
 import "cosmos/cosmos.proto";
 import "cosmos/distribution/distribution.proto";
+import "google/api/annotations.proto";
 
 option go_package = "github.com/cosmos/cosmos-sdk/x/distribution/types";
 
 // Query defines the gRPC querier service for distribution module
 service Query {
   // Params queries params of distribution module
-  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {}
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/cosmos/distribution/v1beta/params";
+  }
 
   // ValidatorOutstandingRewards queries rewards of a validator address
-  rpc ValidatorOutstandingRewards(QueryValidatorOutstandingRewardsRequest) returns (QueryValidatorOutstandingRewardsResponse) {}
+  rpc ValidatorOutstandingRewards(QueryValidatorOutstandingRewardsRequest) returns (QueryValidatorOutstandingRewardsResponse) {
+    option (google.api.http).get = "/cosmos/distribution/v1beta/outstanding-rewards/{validator_address}";
+  }
 
   // ValidatorCommission queries accumulated commission for a validator
-  rpc ValidatorCommission (QueryValidatorCommissionRequest) returns (QueryValidatorCommissionResponse) {}
+  rpc ValidatorCommission (QueryValidatorCommissionRequest) returns (QueryValidatorCommissionResponse) {
+    option (google.api.http).get = "/cosmos/distribution/v1beta/validator-commission/{validator_address}";
+  }
 
   // ValidatorSlashes queries slash events of a validator
-  rpc ValidatorSlashes (QueryValidatorSlashesRequest) returns (QueryValidatorSlashesResponse) {}
+  rpc ValidatorSlashes (QueryValidatorSlashesRequest) returns (QueryValidatorSlashesResponse) {
+    option (google.api.http).get = "/cosmos/distribution/v1beta/validator-slashes/{validator_address}/{starting_height}/{ending_height}";
```

Comment:
Are these `starting_height` and `ending_height` are optional?
```suggestion
    option (google.api.http).get = "/cosmos/distribution/v1beta/validators/{validator_address}/slashes/{starting_height}/{ending_height}";
```

Reply:
They are not optional here, those values can be vary.

Reply:
Should we actually go though all proto `Query*Request` and mark required fields as required?

Reply:
Yes @amaurymartiny, let's address them in a separate PR? This looks big already

Reply:
@atheeshp afiak they should be optional params.

Reply:
Okay